core/metrics-core-service/src/main/java/org/hawkular/metrics/core/dropwizard/HawkularMetricsRegistryListener.java

```diff
@@ -154,4 +275,15 @@ private void createMetrics(String metricName, List<Metric<?>> metrics) {
                 );
     }
 
+    private void deleteMetrics(String metricName, List<MetricId<?>> metrics) {
```

Comment:
This could also be a ``Transformer`` or something else reusable that's then applied to the streams from the ``onMetricType`` methods. Would look more reactive.